kroxylicious-additional-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/TopicNameBasedKekSelector.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+public abstract non-sealed class TopicNameBasedKekSelector<Kr> implements KekSelector {
```

Comment:
Is the intent that each KMS provider providers an implementation of this class (and RecordBasedKekSelector) which will resolve the topicName to Kek using the KMS's underlying API?  Or is it that the Kms interface will expose #findKek() that allow the KMS to be queried in a generic way?    I think it has to be latter because otherwise the kms modules need to depend on the kroxylicious filter module, which seems wrong.